codegen/java-gen/src/main/scala/com/lightbend/akkasls/codegen/java/ViewServiceSourceGenerator.scala

```diff
@@ -75,213 +76,186 @@ object ViewServiceSourceGenerator {
     generatedSources.result()
   }
 
-  private[codegen] def viewRouter(view: ModelBuilder.ViewService, packageName: String): String = {
-    val imports = generateCommandImports(
-      view.commands,
-      view.state,
-      packageName,
-      otherImports = Seq(
-        "com.akkaserverless.javasdk.impl.view.UpdateHandlerNotFound",
-        "com.akkaserverless.javasdk.impl.view.ViewRouter",
-        "com.akkaserverless.javasdk.view.View"))
-
+  private[codegen] def viewRouter(view: ModelBuilder.ViewService, packageName: PackageNaming): String = {
+    import Types.View._
     val cases = view.transformedUpdates
       .map { cmd =>
         val methodName = cmd.name
-        val inputType = qualifiedType(cmd.inputType)
-        s"""|case "$methodName":
+        val inputType = cmd.inputType
+        c"""|case "$methodName":
             |  return view().${lowerFirst(methodName)}(
             |      state,
             |      (${inputType}) event);
-            |""".stripMargin
+            |"""
       }
 
-    s"""package $packageName;
-        |
-        |${writeImports(imports)}
-        |
-        |$managedComment
-        |
-        |/** A view handler */
-        |public class ${view.routerName} extends ViewRouter<${qualifiedType(view.state.fqn)}, ${view.className}> {
-        |
-        |  public ${view.routerName}(${view.className} view) {
-        |    super(view);
-        |  }
-        |
-        |  @Override
-        |  public View.UpdateEffect<${qualifiedType(view.state.fqn)}> handleUpdate(
-        |      String eventName,
-        |      ${qualifiedType(view.state.fqn)} state,
-        |      Object event) {
-        |
-        |    switch (eventName) {
-        |      ${Format.indent(cases, 6)}
-        |
-        |      default:
-        |        throw new UpdateHandlerNotFound(eventName);
-        |    }
-        |  }
-        |
-        |}
-        |""".stripMargin
+    JavaGeneratorUtils.generate(
+      packageName,
+      c"""|$managedComment
+          |
+          |/** A view handler */
+          |public class ${view.routerName} extends $ViewRouter<${view.state.fqn}, ${view.impl}> {
```

Comment:
Maybe even change the `view.routerName` string to a `view.router` FQN everywhere - but not important for this PR.

Reply:
In general it makes sense, but we cannot use FQN directly for name declarations directly because it might start to qualify them in spaces like this where we don't want it.

Reply:
I'm pretty sure it won't ever qualify names that are in the 'current package', but that's indeed a requirement we'd be tied to then.

Reply:
Ah, good point. I guess it would be hard to generate code in a foreign package in the first place. So, I agree the change would be nice but let's fix it later.